tests/gtests/internals/test_sdpa.cpp

```diff
@@ -706,8 +706,29 @@ sdpa_tensors_t get_descriptors(dnnl::engine &eng, const sdpa_dims_t &p,
     return out;
 }
 
+static std::unique_ptr<dnnl::engine> sdpa_eng;
```

Comment:
If you want to get the same instance over and over, it's better to put it inside `get_sdpa_test_engine` and return a `const &` to it.

Reply:
In addition to requiring the same engine repeatedly, the above concern was related to the lifetime of the static outliving the runtime. Moving it into `get_sdpa_test_engine ` returns the lifetime issues while removing access from the TestSuiteSetup/TestSuiteTeardown functions.